criu/config.c

```diff
@@ -592,6 +592,71 @@ static int parse_file_validation_method(struct cr_options *opts, const char *opt
 	return -1;
 }
 
+/*
+ * Parse work_dir, imgs_dir and logfile for early loading of config file.
+ */
+int early_parse_rpc_cfg_file(void)
+{
+	char **_argv = NULL;
+	int _argc = 0;
+	int opt, idx;
+	int rc = 0;
+	int saved_opterr;
+
+	static const char short_opts[] = "D:W:o:";
+	static struct option long_opts[] = {
```

Comment:
Strictly speaking, you can't do that. If you don't know all the available options, you can't distinguish which are options and which are arguments.